pkg/parcacol/querier.go

```diff
@@ -1491,7 +1491,7 @@ func (q *Querier) GetProfileMetadataMappings(
 
 			values := locations.ListValues().(*array.Dictionary)
 
-			compactedDict, err := compactDictionary.CompactDictionary(memory.DefaultAllocator, values)
+			compactedDict, err := compactDictionary.CompactDictionary(q.pool, values)
```

Comment:
I also made the change to use the Querier's memory allocator here instead of `memory.DefaultAllocator`.